src/JsonSchema/ConstraintError.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace JsonSchema;
+
+class ConstraintError extends \MabeEnum\Enum
+{
+    const ADDITIONAL_ITEMS = 'additionalItems';
+    const ADDITIONAL_PROPERTIES = 'additionalProp';
+    const ALL_OF = 'allOf';
+    const ANY_OF = 'anyOf';
+    const DEPENDENCIES = 'dependencies';
+    const DISALLOW = 'disallow';
+    const DIVISIBLE_BY = 'divisibleBy';
+    const ENUM = 'enum';
+    const EXCLUSIVE_MINIMUM = 'exclusiveMinimum';
+    const EXCLUSIVE_MAXIMUM = 'exclusiveMaximum';
+    const FORMAT_COLOR = 'colorFormat';
+    const FORMAT_DATE = 'dateFormat';
+    const FORMAT_DATE_TIME = 'dateTimeFormat';
+    const FORMAT_DATE_UTC = 'dateUtcFormat';
+    const FORMAT_EMAIL = 'emailFormat';
+    const FORMAT_HOSTNAME = 'styleHostName';
+    const FORMAT_IP = 'ipFormat';
+    const FORMAT_PHONE = 'phoneFormat';
+    const FORMAT_REGEX= 'regexFormat';
+    const FORMAT_STYLE = 'styleFormat';
+    const FORMAT_TIME = 'timeFormat';
+    const FORMAT_URL = 'urlFormat';
+    const LENGTH_MAX = 'maxLength';
+    const LENGTH_MIN = 'minLength';
+    const MAXIMUM = 'maximum';
+    const MIN_ITEMS = 'minItems';
+    const MINIMUM = 'minimum';
+    const MISSING_MAXIMUM = 'missingMaximum';
+    const MISSING_MINIMUM = 'missingMinimum';
+    const MAX_ITEMS = 'maxItems';
+    const MULTIPLE_OF = 'multipleOf';
+    const NOT = 'not';
+    const ONE_OF = 'oneOf';
+    const REQUIRED = 'required';
+    const REQUIRED_D3 = 'selfRequired';
+    const REQUIRES = 'requires';
+    const PATTERN = 'pattern';
+    const PREGEX_INVALID = 'pregrex';
+    const PROPERTIES_MIN = 'minProperties';
+    const PROPERTIES_MAX = 'maxProperties';
+    const TYPE = 'type';
+    const UNIQUE_ITEMS = 'uniqueItems';
+
+    public function getMessage()
```

Comment:
May we please also have a `setMessage` method that can add / overwrite messages, and store the messages on the class rather than inside `getMessage`? This would be useful for local translations and people's custom errors that never make it as far as upstream.

Would be good if this could accept either a single error message, or an array of them to allow setting multiple errors at once. e.g.:
```PHP
public static function setMessage($id, $message)
{
    if (is_array($id)) {
        self::$messages = array_merge(self::$messages, $id);
    } else {
        self::$messages[$id] = $message;
    }
}
```